python/lsst/pipe/tasks/fit_coadd_psf.py

```diff
@@ -0,0 +1,207 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "CoaddPsfFitConfig", "CoaddPsfFitSubConfig", "CoaddPsfFitSubTask", "CoaddPsfFitTask",
+]
+
+from .fit_multiband import CatalogExposure
+from lsst.obs.base import ExposureIdInfo
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True, kw_only=True)
+class CatalogExposurePsf(CatalogExposure):
+    def get_catalog(self):
+        return self.catalog
+
+    def get_psf_image(self, source):
+        bbox = source.getFootprint().getBBox()
+        center = bbox.getCenter()
+        return self.exposure.getPsf().computeKernelImage(center).array
+
+
+CoaddPsfFitBaseTemplates = {
+    "name_coadd": "deep",
+    "name_output_method": "multiprofit",
+}
+
+
+class CoaddPsfFitConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap"),
+    defaultTemplates=CoaddPsfFitBaseTemplates,
+):
+    coadd = cT.Input(
+        doc="Coadd image to fit a PSF model to",
+        name="{name_coadd}Coadd_calexp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
+    cat_meas = cT.Input(
+        doc="Deblended single-band source catalog",
+        name="{name_coadd}Coadd_meas",
+        storageClass="SourceCatalog",
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
```

Comment:
You'll also need the scarlet models

Reply:
For PSF fitting? I can't see why. If anything, I'm considering dropping the full exposure and only retrieving the PSF models (I think that's possible).

Reply:
If it's just for the PSF models, why do you need the meas catalog then? Are you just using the columns to identify the un-blended point sources?

Reply:
No, I need source centroids at the minimum, and I also read the standard `detect_isPrimary`/`merge_peak_sky` flags to skip fitting PSF models for those sources since they won't be processed in the source fitting step anyway.